network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -1798,6 +1814,51 @@ public void createSwitches(UUID networkUuid, List<Resource<SwitchAttributes>> re
         return resources;
     }
 
+    public void updateSwitch(UUID networkUuid, Resource<SwitchAttributes> resource) {
+        String kind = resource.getAttributes().getKind() != null ? resource.getAttributes().getKind().toString() : null;
+        ResultSet resultSet = session.execute(psUpdateSwitch.bind(
+                resource.getAttributes().getName(),
+                resource.getAttributes().getProperties(),
+                resource.getAttributes().getNode1(),
+                resource.getAttributes().getNode2(),
+                resource.getAttributes().isOpen(),
+                resource.getAttributes().isRetained(),
+                resource.getAttributes().isFictitious(),
+                kind,
+                resource.getAttributes().getBus1(),
+                resource.getAttributes().getBus2(),
+                networkUuid,
+                resource.getId(),
+                resource.getAttributes().getVoltageLevelId()));
+        if (!resultSet.wasApplied()) {
+            throw new PowsyblException("Switch '" + resource.getId() + "' not updated");
+        }
+    }
+
+    public void updateSwitches(UUID networkUuid, List<Resource<SwitchAttributes>> resources) {
+        for (List<Resource<SwitchAttributes>> subresources : Lists.partition(resources, BATCH_SIZE)) {
+            BatchStatement batch = new BatchStatement(BatchStatement.Type.UNLOGGED);
+            for (Resource<SwitchAttributes> resource : subresources) {
+                String kind = resource.getAttributes().getKind() != null ? resource.getAttributes().getKind().toString() : null;
+                batch.add(creationalStatement(psUpdateSwitch.bind(
```

Comment:
Using a creationalStatement here might not be a good thing, since creationalStatement does not set null values to Cassandra database. creationalStatament should be used only when creating new resources in database.

Reply:
Modification done